### ISSUE TITLE
Children don't keep parent's last name

### ISSUE THREAD

**Issue by twmeehan** [TangoTek/TekTopia-Community#153]
Minor bug:
I thought the children were supposed to keep the last name of their parent. (Maybe I'm just making stuff up) However, the child has a last name that doesn't match anyone in the village.

To Reproduce:
1. Wait for a heart to appear on a villager's bed
2. Place that heart on an empty bed
3. View the child's name to see if it matches anyone

Expected behavior:
Children will keep their parents last name

Screenshots:
![2019-05-28_18 24 19](https://user-images.githubusercontent.com/43288159/58520150-8b0c0e00-8184-11e9-90a7-520c2db421db.png)

Location:
Singleplayer creative:

**Comment by JohnyDL** [comment]
Did their parent die between the heart being made and the child being placed?

**Comment by twmeehan** [comment]
No, I placed the heart as soon as I got it. That was before the parent even left the room

**Comment by JohnyDL** [comment]
Weird. I wonder if it's poling all villagers in all villages not just the local one for a name?

**Comment by TraxKB** [comment]
I would like to add that you can see this in full effect by collecting a bunch of hearts and taking some that stack (same parent) and then placing them in succession; each will have a different surname.

**Comment by Takaarakat** [comment]
Sacred Cows just spawned several children. We had 12 in total, split up into 6 sets of 2 each with the same NBT data. None of the kids had the same last name and I don't think any of the last names matched any of the current villagers. (adding to this listing after removing #820)
